libpod/healthcheck.go

```diff
@@ -278,6 +278,7 @@ func (c *Container) incrementStartupHCSuccessCounter(ctx context.Context) {
 	if recreateTimer {
 		logrus.Infof("Startup healthcheck for container %s passed, recreating timer", c.ID())
 
+		oldUnit := c.state.HCUnitName
```

Comment:
Why do you need a temporary variable for this value?

Reply:
That is the whole point of the bug fix, as written in the commit message createTimer() will overwrite this with a new value thus removeTransientFiles() removed the new unit not the old one.

Maybe I should have added this in a comment there to make it clear